### ISSUE TITLE
Custom Mapping causes "No expression named ___" error when used in table view with joined fields

### ISSUE THREAD

**Issue by jrosenthal** [metabase/metabase#8421]
Your databases: default custom dataset (but also MySQL 5.0)
Metabase version: 0.30.1
Metabase hosting environment: windows, jar (but also CentOS 7, jar)
Metabase internal database: default (but also Postgres)

Custom Mapping throws an error when you use it and and visible fields from a joined table.  

Using the sample dataset and a fresh jar install
- Query Orders
- refresh (loads properly)
- Using table visualization interface, add the product.ein to the viewed fields
- refresh (loads properly)
- change data model for orders so that Quantity is a custom mapped field
- refresh (error - No expression named 'Quantity'.)
- recreate question from scratch (ie: Ask A Question > Custom > Orders table > Add product.ein to fields
- refresh (error - No expression named 'Quantity'.)
- remove custom mapping (change back to value)
- refresh (error, as before)
- recreate question from scratch (ie: Ask A Question > Custom > Orders table > Add product.ein to fields
- refresh (loads properly)

This issue prevents us from upgrading from .29 as it renders many of our queries broken.

Possibly related to: https://github.com/metabase/metabase/issues/8422

Log of Error:
```
Aug 28 20:22:12 WARN metabase.query-processor :: {:status :failed,
:class java.lang.Exception,
:error "No expression named 'Quantity'.",
:stacktrace
["driver.generic_sql.query_processor$expression_with_name.invokeStatic(query_processor.clj:57)"
"driver.generic_sql.query_processor$expression_with_name.invoke(query_processor.clj:53)"
"driver.generic_sql.query_processor$fn__56300.invokeStatic(query_processor.clj:90)"
"driver.generic_sql.query_processor$fn__56300.invoke(query_processor.clj:86)"
"driver.generic_sql.query_processor$apply_fields$iter__56402__56406$fn__56407$fn__56408.invoke(query_processor.clj:229)"
"driver.generic_sql.query_processor$apply_fields$iter__56402__56406$fn__56407.invoke(query_processor.clj:228)"
"driver.generic_sql.query_processor$apply_fields.invokeStatic(query_processor.clj:228)"
"driver.generic_sql.query_processor$apply_fields.invoke(query_processor.clj:225)"
"driver.generic_sql$fn__33627$G__33466__33636.invoke(generic_sql.clj:32)"
"driver.generic_sql.query_processor$apply_clauses.invokeStatic(query_processor.clj:370)"
"driver.generic_sql.query_processor$apply_clauses.invoke(query_processor.clj:365)"
"driver.generic_sql.query_processor$build_honeysql_form.invokeStatic(query_processor.clj:383)"
"driver.generic_sql.query_processor$build_honeysql_form.invoke(query_processor.clj:379)"
"driver.generic_sql.query_processor$mbql__GT_native.invokeStatic(query_processor.clj:391)"
"driver.generic_sql.query_processor$mbql__GT_native.invoke(query_processor.clj:387)"
"driver$fn__28170$G__28065__28177.invoke(driver.clj:104)"
"query_processor.middleware.mbql_to_native$query__GT_native_form.invokeStatic(mbql_to_native.clj:17)"
"query_processor.middleware.mbql_to_native$query__GT_native_form.invoke(mbql_to_native.clj:12)"
"query_processor.middleware.mbql_to_native$mbql__GT_native$fn__33022.invoke(mbql_to_native.clj:26)"
"query_processor.middleware.annotate_and_sort$annotate_and_sort$fn__31336.invoke(annotate_and_sort.clj:42)"
"query_processor.middleware.limit$limit$fn__32977.invoke(limit.clj:15)"
"query_processor.middleware.cumulative_aggregations$cumulative_aggregation$fn__32827.invoke(cumulative_aggregations.clj:58)"
"query_processor.middleware.cumulative_aggregations$cumulative_aggregation$fn__32827.invoke(cumulative_aggregations.clj:58)"
"query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__36951.invoke(results_metadata.clj:51)"
"query_processor.middleware.format_rows$format_rows$fn__32967.invoke(format_rows.clj:26)"
"query_processor.middleware.binning$update_binning_strategy$fn__31430.invoke(binning.clj:165)"
"query_processor.middleware.resolve$resolve_middleware$fn__30924.invoke(resolve.clj:483)"
"query_processor.middleware.expand$expand_middleware$fn__32707.invoke(expand.clj:607)"
"query_processor.middleware.add_row_count_and_status$add_row_count_and_status$fn__31010.invoke(add_row_count_and_status.clj:15)"
"query_processor.middleware.driver_specific$process_query_in_context$fn__32847.invoke(driver_specific.clj:12)"
"query_processor.middleware.resolve_driver$resolve_driver$fn__35494.invoke(resolve_driver.clj:15)"
"query_processor.middleware.bind_effective_timezone$bind_effective_timezone$fn__31344$fn__31345.invoke(bind_effective_timezone.clj:9)"
"util.date$call_with_effective_timezone.invokeStatic(date.clj:82)"
"util.date$call_with_effective_timezone.invoke(date.clj:71)"
"query_processor.middleware.bind_effective_timezone$bind_effective_timezone$fn__31344.invoke(bind_effective_timezone.clj:8)"
"query_processor.middleware.cache$maybe_return_cached_results$fn__31519.invoke(cache.clj:149)"
"query_processor.middleware.catch_exceptions$catch_exceptions$fn__32758.invoke(catch_exceptions.clj:58)"
"query_processor$process_query.invokeStatic(query_processor.clj:135)"
"query_processor$process_query.invoke(query_processor.clj:131)"
"query_processor$run_and_save_query_BANG_.invokeStatic(query_processor.clj:249)"
"query_processor$run_and_save_query_BANG_.invoke(query_processor.clj:243)"
"query_processor$fn__36989$process_query_and_save_execution_BANG___36994$fn__36995.invoke(query_processor.clj:289)"
"query_processor$fn__36989$process_query_and_save_execution_BANG___36994.invoke(query_processor.clj:275)"
"query_processor$fn__37013$process_query_and_save_with_max_BANG___37018$fn__37019.invoke(query_processor.clj:310)"
"query_processor$fn__37013$process_query_and_save_with_max_BANG___37018.invoke(query_processor.clj:306)"
"api.dataset$fn__43466$fn__43469.invoke(dataset.clj:45)"
"api.common$fn__20623$invoke_thunk_with_keepalive__20628$fn__20629$fn__20630.invoke(common.clj:433)"],
:query
{:type "query",
:query
{:source_table 2,
:fields [["field-id" 12] ["field-id" 15] ["field-id" 10] ["field-id" 11] ["field-id" 14] ["field-id" 13] ["field-id" 16] ["field-id" 17] ["field-id" 9] ["expression" "Quantity"] ["fk->" 11 1]]},
:parameters [],
:constraints {:max-results 10000, :max-results-bare-rows 2000},
:info
{:executed-by 1,
:context :ad-hoc,
:card-id nil,
:nested? false,
:query-hash [-83, 24, 72, 50, 35, -77, 88, 20, 117, -105, 61, 57, -104, -120, -7, 83, -40, 54, -2, -34, -118, 42, 56, -41, -109, -46, 96, -10, -60, 14, -66, 122],
:query-type "MBQL"}},
:expanded-query nil}

Aug 28 20:22:12 WARN metabase.query-processor :: Query failure: No expression named 'Quantity'.
["query_processor$assert_query_status_successful.invokeStatic(query_processor.clj:217)"
"query_processor$assert_query_status_successful.invoke(query_processor.clj:210)"
"query_processor$run_and_save_query_BANG_.invokeStatic(query_processor.clj:250)"
"query_processor$run_and_save_query_BANG_.invoke(query_processor.clj:243)"
"query_processor$fn__36989$process_query_and_save_execution_BANG___36994$fn__36995.invoke(query_processor.clj:289)"
"query_processor$fn__36989$process_query_and_save_execution_BANG___36994.invoke(query_processor.clj:275)"
"query_processor$fn__37013$process_query_and_save_with_max_BANG___37018$fn__37019.invoke(query_processor.clj:310)"
"query_processor$fn__37013$process_query_and_save_with_max_BANG___37018.invoke(query_processor.clj:306)"
"api.dataset$fn__43466$fn__43469.invoke(dataset.clj:45)"
"api.common$fn__20623$invoke_thunk_with_keepalive__20628$fn__20629$fn__20630.invoke(common.clj:433)"]

Aug 28 20:22:12 DEBUG metabase.middleware :: POST /api/dataset 200 (516 ms) (12 DB calls). Jetty threads: 8/50 (4 busy, 6 idle, 0 queued)```

**Comment by flamber** [comment]
Hi @jrosenthal 

I cannot reproduce your issue on 0.31.2 - please test with that version to see if this issue is resolved.

I never get any errors, but when setting `QUANTITY` to "Custom mapping" and pressing "Save", then the result returns an empty column:
![image](https://user-images.githubusercontent.com/1447303/53207171-a224e800-35f8-11e9-90b9-52a0d6739fec.png)

But when I change "Field Type" from "Quantity" to "No special type", then it displays my mappings:
![image](https://user-images.githubusercontent.com/1447303/53207015-2e82db00-35f8-11e9-8d6c-dda38428218d.png)

**Comment by camsaul** [comment]
This should be fixed in 0.32.9

**Comment by flamber** [comment]
Closing since I have not been able to reproduce since 0.31.2 - latest release is 0.34.3
